src/ui/features/xp-drop/XpDropClaimFlow/XpDropClaim.tsx

```diff
@@ -0,0 +1,526 @@
+    mutationFn: (address: string) => setCurrentAddress({ address }),
+    onSuccess() {
+      refetchCurrentWallet();
+    },
+  });
+
+  const [selectedWallet, setSelectedWallet] = useState(currentWallet);
+  const [claimedInfo, setClaimedInfo] = useState<{
+    address: string;
+    meta: WalletMeta | null;
+  } | null>(null);
+
+  const { data: walletGroups, ...walletGroupsQuery } = useWalletGroups();
+  const { signerWalletGroups, signerWalletAddresses } = useMemo(() => {
+    const signerWalletGroups =
+      walletGroups?.filter((group) =>
+        isSignerContainer(group.walletContainer)
+      ) ?? [];
+    const signerWalletAddresses = signerWalletGroups
+      .flatMap((group) => group.walletContainer.wallets)
+      .map((wallet) => wallet.address);
+    return { signerWalletGroups, signerWalletAddresses };
+  }, [walletGroups]);
+
+  const { data: walletsMeta, ...walletsMetaQuery } = useQuery({
+    queryKey: ['ZerionAPI.getWalletsMeta', signerWalletAddresses],
+    queryFn: () => getWalletsMetaByChunks(signerWalletAddresses),
+  });
+
+  const { eligibleWalletGroups, eligibleAddresses } = useMemo(() => {
+    const eligibleAddresses =
+      walletsMeta
+        ?.filter((meta) => Boolean(meta.membership.retro))
+        .map((meta) => normalizeAddress(meta.address)) ?? [];
+
+    const eligibleWalletGroups = signerWalletGroups
+      ?.map((group) => ({
+        id: group.id,
+        walletContainer: {
+          wallets: group.walletContainer.wallets.filter((wallet) => {
+            const normalizedAddress = normalizeAddress(wallet.address);
+            return (
+              eligibleAddresses.includes(normalizedAddress) &&
+              normalizedAddress !== claimedInfo?.address
+            );
+          }),
+        },
+      }))
+      .filter((group) => group.walletContainer.wallets.length > 0);
+
+    return { eligibleWalletGroups, eligibleAddresses };
+  }, [signerWalletGroups, walletsMeta, claimedInfo]);
+
+  const signMsgBtnRef = useRef<SignMsgBtnHandle | null>(null);
+  const walletSelectDialogRef = useRef<HTMLDialogElementInterface | null>(null);
+  const xpBreakdownDialogRef = useRef<HTMLDialogElementInterface | null>(null);
+
+  const selectedWalletMeta = useMemo(
+    () =>
+      selectedWallet
+        ? walletsMeta?.find(
+            (meta) =>
+              normalizeAddress(meta.address) ===
+              normalizeAddress(selectedWallet.address)
+          )
+        : null,
+    [selectedWallet, walletsMeta]
+  );
+
+  const { mutate: personalSign, ...personalSignMutation } = useMutation({
+    mutationFn: async () => {
+      invariant(selectedWallet, 'selectedWallet is required');
+      invariant(signMsgBtnRef.current, 'SignMessageButton not found');
+
+      const message = 'I want to claim my Zerion Retro XP';
+      const signature = await signMsgBtnRef.current.personalSign({
+        params: [message],
+        initiator: INTERNAL_ORIGIN,
+        clientScope: null,
```

Comment:
As far as I understand this is needed by analytics: 
```suggestion
        clientScope: 'Claim XP',
```